mesop/tests/e2e/update_page_title_test.ts

```diff
@@ -0,0 +1,119 @@
+import {Page, test, expect} from '@playwright/test';
+
+const expectTitleUpdate = async (page: Page, title: string) => {
+  return await expect(async () => {
+    expect(await page.title()).toEqual(title);
+  }).toPass({timeout: 5000});
+};
+
+test.describe('Update page title', () => {
+  test('title updates', async ({page}) => {
+    await page.goto('/update_page_title/page1');
+
+    await page.getByText('update title').click();
+    await expectTitleUpdate(page, 'Update title page 1');
+
+    // Title should stay the same even though state is updated.
+    await page.getByText('title does not change if clicked').click();
+    await expectTitleUpdate(page, 'Update title page 1');
+  });
+
+  test.describe('navigation with separate yields', () => {
+    test('title change before navigate uses page title', async ({page}) => {
+      await page.goto('/update_page_title/page1');
+
+      await page
+        .getByRole('button', {
+          name: 'title change before navigate',
+          exact: true,
+        })
+        .click();
+      await expect(page).toHaveURL(
+        'http://127.0.0.1:32123/update_page_title/page2',
```

Comment:
I think you can drop the origin, see: https://github.com/google/mesop/blob/1f34f5669ba2ab064d91a6e11e1da5848bf92077/mesop/tests/e2e/dom_state_test.ts#L13